source/shared/version.h

```diff
@@ -47,13 +45,22 @@
 // Main version, dot separated 3 digits, Major.Minor.Patch
 #define VER_APIVERSION_STR      STRINGIFY( SQLVERSION_MAJOR ) "." STRINGIFY( SQLVERSION_MINOR ) "." STRINGIFY( SQLVERSION_PATCH )
 
-// For preview release, we want the following:
-// #define VER_FILEVERSION_STR     VER_APIVERSION_STR "-" SEMVER_PRERELEASE SEMVER_BUILDMETA
-// because pecl doesn't like dashes. However, if SEMVER_PRERELEASE is empty, the "-" must be removed
+// Semantic versioning: 
+// For stable releases leave SEMVER_PRERELEASE empty
+// Otherwise, for pre-releases, add '-' and change it to:
+// "RC" for release candidates
+// "preview" for ETP 
+#if PREVIEW > 0
+#undef SEMVER_PRERELEASE
+#define SEMVER_PRERELEASE "preview"
+#define VER_FILEVERSION_STR     VER_APIVERSION_STR "-" SEMVER_PRERELEASE SEMVER_BUILDMETA
+#else
```

Comment:
SEMVER_BUILDMETA is empty unless SQLVERSION_BUILD is not 0, but when is SQLVERSION_BUILD not zero?

Reply:
The build scripts take care of setting the SQLVERSION_BUILD, see this line

https://github.com/Microsoft/msphpsql/blob/master/buildscripts/buildtools.py?utf8=%E2%9C%93#L269